workers/generic-worker/process/multiuser_posix.go

```diff
@@ -11,23 +11,23 @@ import (
 	"strconv"
 	"syscall"
 
-	"github.com/taskcluster/taskcluster/v69/workers/generic-worker/runtime"
+	gwruntime "github.com/taskcluster/taskcluster/v69/workers/generic-worker/runtime"
 	"golang.org/x/net/context"
 )
 
 type PlatformData struct {
 	SysProcAttr *syscall.SysProcAttr
 }
 
-func NewPlatformData(currentUser bool) (pd *PlatformData, err error) {
+func NewPlatformData(currentUser bool, headlessTasks bool, user *gwruntime.OSUser) (pd *PlatformData, err error) {
 	if currentUser {
 		return &PlatformData{}, nil
 	}
-	return TaskUserPlatformData()
+	return TaskUserPlatformData(user, headlessTasks)
 }
 
-func TaskUserPlatformData() (pd *PlatformData, err error) {
-	username, err := runtime.InteractiveUsername()
+func TaskUserPlatformData(u *gwruntime.OSUser, headlessTasks bool) (pd *PlatformData, err error) {
+	username, err := gwruntime.InteractiveUsername()
```

Comment:
This is still depending on gdm as it ends up trying to get the username with 

`gnomeSessionUserList, err := host.CombinedOutput("/usr/bin/env", "bash", "-c", "PROCPS_USERLEN=20 /usr/bin/w | /bin/grep gnome-[s]ession | /usr/bin/cut -f1 -d' '")`

What I did on my attempt to solve this issue was to copy the code from `gdm3/gdm3_freebsd.go` and ended up with something like that:

```go
func TaskUserPlatformData(config *gwconfig.Config) (pd *PlatformData, err error) {
    var user string
    if !config.DisableGui {
        user, err = runtime.InteractiveUsername()
        if err != nil {
            return nil, fmt.Errorf("could not determine interactive username: %v", err)
        }
    } else {
        // TODO: Dedupe this block, it's also in gdm3/freebsd
        credsFile, err := os.Open("current-task-user.json")
        if err != nil {
            return nil, err
        }
        defer func() {
            credsFile.Close()
        }()
        decoder := json.NewDecoder(credsFile)
        decoder.DisallowUnknownFields()
        decodedUser := map[string]string{}
        err = decoder.Decode(&decodedUser)
        if err != nil {
            return nil, err
        }
        user = decodedUser["name"]
    }
```

Reply:
Ah whoops, thanks for that! I think there is a simple fix:

```
diff --git a/workers/generic-worker/process/multiuser_posix.go b/workers/generic-worker/process/multiuser_posix.go
index a95d67f9b..054f30a4d 100644
--- a/workers/generic-worker/process/multiuser_posix.go
+++ b/workers/generic-worker/process/multiuser_posix.go
@@ -27,12 +27,7 @@ func NewPlatformData(currentUser bool, headlessTasks bool, user *gwruntime.OSUse
 }
 
 func TaskUserPlatformData(u *gwruntime.OSUser, headlessTasks bool) (pd *PlatformData, err error) {
-       username, err := gwruntime.InteractiveUsername()
-       if err != nil {
-               return nil, fmt.Errorf("could not determine interactive username: %w", err)
-       }
-
-       usr, err := user.Lookup(username)
+       usr, err := user.Lookup(u.Name)
        if err != nil {
                return nil, fmt.Errorf("failed to lookup user: %w", err)
        }
```